cmd/xl-sets.go

```diff
@@ -304,15 +305,22 @@ func newXLSets(endpoints EndpointList, format *formatXLV3, setCount int, drivesP
 // StorageInfo - combines output of StorageInfo across all erasure coded object sets.
 func (s *xlSets) StorageInfo(ctx context.Context) StorageInfo {
 	var storageInfo StorageInfo
+	var wg sync.WaitGroup
 	storageInfo.Backend.Type = BackendErasure
 	for _, set := range s.sets {
-		lstorageInfo := set.StorageInfo(ctx)
-		storageInfo.Used = storageInfo.Used + lstorageInfo.Used
-		storageInfo.Total = storageInfo.Total + lstorageInfo.Total
-		storageInfo.Available = storageInfo.Available + lstorageInfo.Available
-		storageInfo.Backend.OnlineDisks = storageInfo.Backend.OnlineDisks + lstorageInfo.Backend.OnlineDisks
-		storageInfo.Backend.OfflineDisks = storageInfo.Backend.OfflineDisks + lstorageInfo.Backend.OfflineDisks
-	}
+		wg.Add(1)
+		go func(set *xlObjects) {
+			defer wg.Done()
+			lstorageInfo := set.StorageInfo(ctx)
```

Comment:
This type change is not needed, also no need to use atomic - just capture storageInfo as a list of sets in parallel and combine them later. We can make it lock free.

Reply:
ok